core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -166,13 +173,41 @@ private[spark] class DiskBlockManager(
     }
   }
 
+  /**
+   * SPARK-37618: Makes sure that the file is created as world readable. This is to get
+   * around the fact that making the block manager sub dirs group writable removes
+   * the setgid bit in secure Yarn environments, which prevents the shuffle service
+   * from being able ot read shuffle files. The outer directories will still not be
+   * world executable, so this doesn't allow access to these files except for the
+   * running user and shuffle service.
+   */
+  def createWorldReadableFile(file: File): Unit = {
```

Comment:
See comment on getFile above, with that, we dont need `createWorldReadableFile` or `createTempFileWith` (below) ?

Reply:
See response on that above